test/test_optim.py

```diff
@@ -2149,7 +2151,8 @@ def _test_against_closed_form(self, scheduler, closed_form_scheduler, epochs=10)
         targets = []
         for epoch in range(epochs):
             closed_form_scheduler.optimizer.step()
-            closed_form_scheduler.step(epoch)
+            with warnings.catch_warnings(record=True) as w:
```

Comment:
Same this one should not warn?

Reply:
No worries about the confusion -- I relate to it too 😅 

In both cases the warning comes from passing in the epoch parameter -- which triggers this warning. https://github.com/pytorch/pytorch/blob/191b48b12f33e1e9525882da0c62b68686d69e42/torch/optim/lr_scheduler.py#L154

So I think that we ought to expect the warning to appear in all cases since we are making sure that the deprecated behaviour is functional for as long as it remains in the codebase

Reply:
Ho ok! Sounds good! Thanks for checking.